tests/integration/kubernetes/confidential_kbs.sh

```diff
@@ -154,6 +154,20 @@ kbs_set_resource_from_file() {
 	kbs-client --url "$(kbs_k8s_svc_http_addr)" config \
 		--auth-private-key "$KBS_PRIVATE_KEY" set-resource \
 		--path "$path" --resource-file "$file"
+
+	kbs_pod=$(kubectl -n coco-tenant get pods -o NAME)
+	kbs_repo_path="/opt/confidential-containers/kbs/repository"
+	for i in {1..3}
+	do
+		sleep 5
+		kubectl -n coco-tenant exec -it "$kbs_pod" -- \
+			bash -c "ls $kbs_repo_path/$path" && \
```

Comment:
Not sure how important but imagine one sets a `path`, then one changes the same `path` to a different value. In that case this returns immediately as the `path` already exists but it still contains the previous value. It's hypothetical only so let me know what you think...

Reply:
I think this doesn't match the case in our tests, so I'm fine not taking this into consideration.

Reply:
Actually how about changing the loop from the controller to the pod? I think it'd consume less resources so we could even afford doing busy loop (something like: `kubectl -n coco-tenant exec -it "$kbs_pod" -- bash -c "for i in {1..30}; do [ -e '$kbs_repo_path/$path' ] && exit 0; done exit -1`"

Reply:
^^ yep, I was just a bit paranoiac :-) Anyway what do you think about the for loop inside the container? I'm not particularly happy about the 5s interval and moving the logic there should allow decreasing the sleep time...

Reply:
I sincerely don't mind, but is it worth it?  What's your issue with the 5s interval? 5s added to the 50 minutes (3000s) the CI takes to finish this run?

Reply:
Also, I think we lose important debug information by doing that, and in my mind that's more important than the 5s waiting, mainly when something goes wrong.

Reply:
As for the debugability it'd definitely require more post-info gathering. This was just a base example.

As for the speed, I tend to prefer busier loops to avoid things working only because there was sufficient timeout. If we add 5s timeout after everything, things usually work. But what if someone won't wait and it breaks somewhere? So my motivation is not to speed-up the testing, my motivation is to start testing as soon as possible.

All in all I'm not against this, it's definitely safer. I'd slightly prefer busier loop with decent post-failure info to minimal 5s wait between each resource set.

Reply:
Note that on my machine this usually takes 0s, which is why I was motivated to speed things up a bit...

Reply:
I have added a new commit changing the loop, and I'd prefer having that as a separate commit.
I am sincerely not happy with the change, at least not as the person who'll end up debugging this, but it is what it is, and I'm following the suggestion.